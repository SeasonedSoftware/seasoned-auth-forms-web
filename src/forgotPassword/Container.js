import React from 'react'
import { Formik } from 'formik'
import PropTypes from 'prop-types'

import ForgetPassword from './Form'
import validate from '../validations/validate'

const Container = ({ onSubmit, ...props }) => (
  <Formik
    initialValues={{
      email: '',
    }}
    validate={validate({ password: false, ...props })}
    onSubmit={onSubmit}
    render={form => <ForgetPassword {...props} form={form} />}
  />
)

Container.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
}
export default Container
