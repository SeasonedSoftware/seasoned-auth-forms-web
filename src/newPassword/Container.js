import React from 'react'
import { Formik } from 'formik'
import PropTypes from 'prop-types'

import NewPassword from './Form'
import validate from '../validations/validate'

const Container = ({ onSubmit, ...props }) => (
  <Formik
    initialValues={{
      password: '',
      passwordConfirmation: '',
    }}
    validate={validate({ options: { newPassword: true }, ...props })}
    onSubmit={onSubmit}
    render={form => <NewPassword {...props} form={form} />}
  />
)

Container.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
}
export default Container
