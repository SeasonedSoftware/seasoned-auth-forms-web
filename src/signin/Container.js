import React from 'react'
import { Formik } from 'formik'
import PropTypes from 'prop-types'

import Form from './Form'
import validate from './validate'

const SignIn = ({ onSubmit, ...props }) => (
  <Formik
    initialValues={{
      email: '',
      password: '',
    }}
    validate={validate(props)}
    onSubmit={onSubmit}
    render={form => <Form {...props} form={form} />}
  />
)

SignIn.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
}
export default SignIn
