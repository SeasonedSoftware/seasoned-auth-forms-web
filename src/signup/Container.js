import React from 'react'
import { Formik } from 'formik'
import PropTypes from 'prop-types'

import Form from './Form'
import validate from '../validations/validate'

const SignUp = ({ onSubmit, ...props }) => (
  <Formik
    initialValues={{
      email: '',
      password: '',
      passwordConfirmation: '',
    }}
    validate={validate({ newPassword: true, ...props })}
    onSubmit={onSubmit}
    render={form => <Form {...props} form={form} />}
  />
)

SignUp.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
}

export default SignUp
