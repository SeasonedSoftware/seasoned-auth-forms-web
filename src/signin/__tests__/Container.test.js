import React from 'react'
import renderer from 'react-test-renderer'

import Container from '../Container'

jest.mock('formik', () => ({
  Formik: props => (
    <div {...props}>Formik - {props.render({ foo: 'bar' })}</div>
  ),
}))

jest.mock('../Form', () => props => <div {...props}>Form</div>)

it('renders correctly', () => {
  const props = { onSubmit: jest.fn() }
  const tree = renderer.create(<Container {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
