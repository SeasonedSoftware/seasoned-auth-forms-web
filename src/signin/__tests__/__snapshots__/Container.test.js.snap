// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  initialValues={
    Object {
      "email": "",
      "password": "",
    }
  }
  onSubmit={[MockFunction]}
  render={[Function]}
  validate={[Function]}
>
  Formik - 
  <div
    form={
      Object {
        "foo": "bar",
      }
    }
  >
    Form
  </div>
</div>
`;
